modules/k6/k6.go

```diff
@@ -0,0 +1,95 @@
+package k6
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// K6Container represents the K6 container type used in the module
+type K6Container struct {
+	testcontainers.Container
+}
+
+// WithTestScript mounts the given script into the ./test directory in the container
+// and passes it to k6 as the test to run.
+// The path to the script must be an absolute path
+func WithTestScript(scriptPath string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		script := filepath.Base(scriptPath)
+		target := fmt.Sprintf("/tests/%s", script)
+		mount := testcontainers.ContainerMount{
+			Source: testcontainers.GenericBindMountSource{
+				HostPath: scriptPath,
+			},
+			Target: testcontainers.ContainerMountTarget(target),
+		}
+		req.Mounts = append(req.Mounts, mount)
+		req.Cmd = append(req.Cmd, target)
+	}
+}
+
+// WithCmdOptions pass the given options to the k6 run command
+func WithCmdOptions(options ...string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Cmd = append(req.Cmd, options...)
+	}
+}
+
+// SetEnvVar sets an environment variable for the test script
+func SetEnvVar(variable string, value string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Cmd = append(req.Cmd, "--env", fmt.Sprintf("%s=%s", variable, value))
+	}
+}
+
+// WithCache uses the given directory as a cache directory building the k6 binary.
+// The path to the directory must be an absolute path
+// Note: The container must run using an user that
+// has access to the directory. See AsUser option
+func WithCache(cacheDir string) testcontainers.CustomizeRequestOption {
```

Comment:
I wonder if we need to be mounted in the host. Do we need to access the cache from outside the container for any reason? Asking as a newbie in k6

Reply:
In reality we don't need access to the cache from outside the container. 

The main reason for mounting is that mounting a local directory makes it easier for the test to create a temporary cache dir and then clean it up when the tests are finished. Please notice that as mentioned in another comment, the cache is intended to be used not only by one test but across multiple tests in a test suite. 

The alternative I considered was using a volume but as far as I can see, TestContainers do not offer any API to clean up this volume once the tests are finished.

Reply:
Ok, now I get the reasons why we need the cache: to speed up "successive" executions, not the current one. Cool.

Have you considered using a kind of a singleton container for this? Where we start the container once and then use it across all the tests in a suite?

Reply:
>Have you considered using a kind of a singleton container for this? Where we start the container once and then use it across all the tests in a suite?

yes, but I think that makes the interface more complex because each successive execution may require passing a different test script.  

I think providing a way of having a cache across executions has better usability. 

The only drawback of the current implementation is the need to set the container user. 

That is why lean towards using a volume, but I don't see an easy way to remove it when the test suite ends.

Reply:
Have you also taken a look to reusable containers? https://golang.testcontainers.org/features/creating_container/#reusable-container

Reply:
I think that would require a complete redesign of the module, as now the module executes a command and ends. There's no way to keep it waiting to execute other scripts. 

If passing the cache so problematic, maybe I should simply remove the feature at the expense of a significant penalty in the test execution time.

Reply:
> We are considering adding volume support for an eventual v1 release, where you could manage volumes with testcontainers APIs. That would simplify the thing.

That would be an interesting addition

> In the mean time, I'd do the following:
>
1. create a named volume using Docker cli, directly.

That makes sense. I will change the implementation for using a volume as cache, add some examples and remove the `AsUser` option which is no longer needed. 

> please change how the test scripts are copied: instead of Mounts, please use Files instead.

This is already done

Reply:
By the way, the problem is not creating the volume. The volume is created if it doesn't exist. The real problem is to clean it up after the test suite ends.

Reply:
If you manually set the right labels in the new volume, then Ryuk will remove it: https://github.com/testcontainers/moby-ryuk/blob/main/main.go#L334

You can get the labels from `testcontainersdocker.DefaultLabels(testcontainerssession.SessionID())` (_Just realised how friendly is that 🤦_)

Reply:
I think we should add this as a helper function in a follow-up PR (I volunteer for that). Meanwhile, I will simply document that it is the responsibility of the test to cleanup the volume when not longer needed.